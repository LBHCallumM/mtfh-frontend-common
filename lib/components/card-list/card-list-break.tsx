import React from "react";

export const CardListBreak = () => {
  return <hr className="mtfh-card-list-break" />;
};
