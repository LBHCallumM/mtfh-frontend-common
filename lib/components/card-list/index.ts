export * from "./card-list";
export * from "./card-list-break";
export * from "./card-list-item";
export * from "./card-list-rows";
