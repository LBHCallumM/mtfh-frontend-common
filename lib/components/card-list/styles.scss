@import "~lbh-frontend/lbh/base";
@import "~govuk-frontend/govuk/helpers/media-queries";

.mtfh-card-list {
  display: grid;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  gap: 1rem;
  align-items: flex-start;
  overflow: hidden;
  margin-bottom: 2rem;

  @include govuk-media-query($from: tablet) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }

  @include govuk-media-query($from: desktop) {
    grid-template-columns: repeat(3, minmax(0, 1fr));
  }

  @include govuk-media-query($from: 992px) {
    grid-template-columns: repeat(4, minmax(0, 1fr));
  }

  > * {
    margin: 0;
  }

  &-item {
    padding: 10px;
    border: 1px solid rgba(173, 171, 171, 0.4);
    border-radius: 3px;

    > * {
      margin: 6px 0 0;
    }

    a {
      @include lbh-heading-h4;
      margin: 0;
    }
  }

  &-rows {
    margin: 10px 0;

    &__row {
      display: flex;
      margin-top: 0;
    }

    &__label {
      margin-top: 0;
      font-weight: $lbh-font-weight-bold;
      margin-right: 5px;
    }

    &__value {
      margin-top: 0;
      color: lbh-colour("lbh-grey-1");
    }
  }

  &-break {
    border: none;
    border-bottom: 1px solid #bfc1c3;
    margin: 10px 0;
    width: 100%;
    max-width: 160px;
  }
}
