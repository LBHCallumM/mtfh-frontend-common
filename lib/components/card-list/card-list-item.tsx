import React, { ComponentPropsWithoutRef } from "react";
import cn from "classnames";

import "./styles.scss";

export const CardListItem = ({
  className,
  children,
}: ComponentPropsWithoutRef<"div">) => {
  return <div className={cn("mtfh-card-list-item", className)}>{children}</div>;
};
