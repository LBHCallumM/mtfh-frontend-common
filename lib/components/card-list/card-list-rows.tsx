import React from "react";
import "./styles.scss";

export type CardListRow = {
  label: string;
  value: string;
};

export interface CardListRowsProp {
  rows: CardListRow[];
}

export const CardListRows = ({ rows }: CardListRowsProp) => {
  return (
    <dl className="mtfh-card-list-rows">
      {rows.map((row: CardListRow, index) => (
        <div key={index} className="mtfh-card-list-rows__row">
          <dt className="mtfh-card-list-rows__label">{row.label}:</dt>
          <dd className="mtfh-card-list-rows__value">{row.value}</dd>
        </div>
      ))}
    </dl>
  );
};
