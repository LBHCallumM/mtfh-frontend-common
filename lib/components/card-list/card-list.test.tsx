import React from "react";
import { render } from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";
import { CardListBreak } from ".";
import { CardList } from "./card-list";
import { CardListItem } from "./card-list-item";
import { CardListRows } from "./card-list-rows";

test("CardList renders", () => {
  const rows = [
    { value: "value1", label: "Label 1" },
    { value: "value2", label: "Label 2" },
  ];
  const { container } = render(
    <CardList>
      <CardListItem>
        <CardListRows rows={rows} />
        <CardListBreak />
        <p>Additional content</p>
      </CardListItem>
    </CardList>,
  );
  expect(screen.getByText("value1")).toBeInTheDocument();
  expect(screen.getByText("Label 1:")).toBeInTheDocument();
  expect(screen.getByText("value2")).toBeInTheDocument();
  expect(screen.getByText("Label 2:")).toBeInTheDocument();
  expect(screen.getByText("Additional content")).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});
