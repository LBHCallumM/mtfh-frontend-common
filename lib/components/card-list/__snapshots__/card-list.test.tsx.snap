// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList renders 1`] = `
<div>
  <div
    class="mtfh-card-list"
  >
    <div
      class="mtfh-card-list-item"
    >
      <dl
        class="mtfh-card-list-rows"
      >
        <div
          class="mtfh-card-list-rows__row"
        >
          <dt
            class="mtfh-card-list-rows__label"
          >
            Label 1
            :
          </dt>
          <dd
            class="mtfh-card-list-rows__value"
          >
            value1
          </dd>
        </div>
        <div
          class="mtfh-card-list-rows__row"
        >
          <dt
            class="mtfh-card-list-rows__label"
          >
            Label 2
            :
          </dt>
          <dd
            class="mtfh-card-list-rows__value"
          >
            value2
          </dd>
        </div>
      </dl>
      <hr
        class="mtfh-card-list-break"
      />
      <p>
        Additional content
      </p>
    </div>
  </div>
</div>
`;
